src/parser/jobs/brd/Barrage.js

```diff
@@ -0,0 +1,499 @@
+
+				// DPS loss is the difference between potential DPS and total DPS (duh)
+				let dpsLoss = this.util.formatDecimal(potentialDPS - totalDPS)
+
+				// If this was a badBarrage
+				if (BAD_ST_WEAPONSKILLS.includes(barrage.skillBarraged)) {
+
+					// Potential damage and DPS loss then become an interval, because a badBarrage could've been either Empyreal Arrow or Refulgent Arrow
+					potentialDamage = `${potentialEmpyrealDamage} - ${potentialRefulgentDamage}`
+					dpsLoss = `${this.util.formatDecimal(potentialEmpyrealDPS - totalDPS)} - ${this.util.formatDecimal(potentialRefulgentDPS- totalDPS)}`
+
+					// Adds the {issue, severity, reason} tuple corresponding a badBarrage to the panel
+					panel.headers.push({
+						issue: <>
+							This barrage was <strong>not</strong> used on <ActionLink {...ACTIONS.EMPYREAL_ARROW}/> or <ActionLink {...ACTIONS.REFULGENT_ARROW}/>.
+						</>,
+						severity: ERROR,
+						reason: <>
+							Your two strongest weaponskills are <ActionLink {...ACTIONS.EMPYREAL_ARROW} /> and <ActionLink {...ACTIONS.REFULGENT_ARROW} />. Make sure you only use your {ACTIONS.BARRAGE.name} on these two skills.
+						</>,
+					})
+				}
+
+				// Only adds the DPS loss table as a custom content in case it was any case of bad barrage (badBarrage, unalignedBarrage, worstBarrageNA)
+				// That's what the first 'if' is for
+				panel.contents.push(<>
+					<Table collapsing unstackable celled>
+						<Table.Header>
+							<Table.Row>
+								<Table.HeaderCell width={2}>
+									<strong>Total damage</strong>
+								</Table.HeaderCell>
+								<Table.HeaderCell width={2}>
+									<strong>Potential damage</strong>
+								</Table.HeaderCell>
+								<Table.HeaderCell width={2} negative>
+									<strong>DPS loss</strong>
+								</Table.HeaderCell>
+							</Table.Row>
+						</Table.Header>
+						<Table.Body>
+							<Table.Row>
+								<Table.Cell>{totalDamage}</Table.Cell>
+								<Table.Cell>{potentialDamage}</Table.Cell>
+								<Table.Cell negative>{dpsLoss}</Table.Cell>
+							</Table.Row>
+						</Table.Body>
+					</Table>
+				</>)
+			}
+
+			// Then builds the panel and returns it in the mapping function
+			return panel.build()
+
+		})
+
+		// Output is an Accordion made with panels, one for each barrage event
+		return <Accordion
+			exclusive={false}
+			panels={panels}
+			styled
+			fluid
+		/>
+	}
+}
+
+// BarrageEvent is an object used to describe a Barrage.
+// Contains information about the cast event, the damage events, the id of the barraged skill, and alignment status with Raging Strikes
+
+class BarrageEvent {
+
+	castEvent = null
+
+	skillBarraged = null
+	damageEvents = []
+
+	// Assuming barrage was aligned with Raging Strikes. Will set to false if determined otherwise
+	aligned = true
+
+	get timestamp() {
+		return this.castEvent && this.castEvent.timestamp
+	}
+}
+
+// BarrageRule is an extention of TieredRule, with the following additions:
```

Comment:
Oh boy. Here we go.

I'm not happy w/ the idea of hidden values at all. Even if it won't technically happen in this use case, it's setting precedent. What if someone gets a top-level thumbs-down but every requirement looks A-OK? The final % should be directly representing the content shown on expansion.

I'm less wary of the `reverse`, but not super on board with it.

Weights are a bit iffy, they become brittle as they always need to add up to 1. If you're absolutely _determined_ to have weightings, an implementation that isn't bounded by this should be used.